libsolidity/analysis/TypeChecker.cpp

```diff
@@ -445,17 +447,32 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 		if (parameterCountMessage.has_value())
 			m_errorReporter.typeError(9128_error, _function.parameterList().location(), parameterCountMessage.value());
 		else if (_function.parameterList().parameters().size() == 2)
-			if (
-				auto const* exponentType = dynamic_cast<IntegerType const*>(_function.parameterList().parameters().at(1)->type());
-				exponentType && exponentType->isSigned()
-			)
+		{
+			auto const* mantissaType = dynamic_cast<IntegerType const*>(_function.parameterList().parameters().at(0)->type());
+			auto const* exponentType = dynamic_cast<IntegerType const*>(_function.parameterList().parameters().at(1)->type());
+
+			vector<string> mantissaOrExponentTypeErrorMessages;
+			if (!mantissaType)
+				mantissaOrExponentTypeErrorMessages.emplace_back("The mantissa parameter must be an integer");
+			if (!exponentType)
+				mantissaOrExponentTypeErrorMessages.emplace_back("The exponent parameter must be an unsigned integer");
+
+			if (!mantissaOrExponentTypeErrorMessages.empty())
+				m_errorReporter.typeError(
+					1587_error,
+					_function.parameterList().location(),
+					"Literal suffix function has invalid parameter types. " +
+					joinHumanReadable(mantissaOrExponentTypeErrorMessages, ".") + "."
```

Comment:
This joins without adding a space:
```
// TypeError 1587: (15-47): Literal suffix function has invalid parameter types. The mantissa parameter must be an integer.The exponent parameter must be an unsigned integer.
```

I'd do it like this:
```suggestion
					joinHumanReadable(mantissaOrExponentTypeErrorMessages, " ")
```
and then I'd include the dot at the end of each message.

IIRC `" "` is the default so you may even just omit that.